runtime/cudaq/qis/qubit_qis.h

```diff
@@ -306,6 +306,25 @@ inline void exp_pauli(double theta, const char *pauliWord,
                                spin_op::from_word(pauliWord));
 }
 
+/// @brief Apply a general Pauli rotation with control qubits and a variadic set
+/// of qubits. The number of qubits must be equal to the pauli word length.
+template <typename QuantumRegister, typename... QubitArgs>
+  requires(std::ranges::range<QuantumRegister>)
+inline void exp_pauli(QuantumRegister &ctrls, double theta,
```

Comment:
nit: the `inline` is not required with a template function (but it's fine either way)